pkg/driver/aci_driver.go

```diff
@@ -0,0 +1,770 @@
+	subscriptionID     string
+	out                io.Writer
+}
+
+// Config returns the ACI driver configuration options
+func (d *ACIDriver) Config() map[string]string {
+	return map[string]string{
+		"VERBOSE":                  "Increase verbosity. true, false are supported values",
+		"AZURE_CLIENT_ID":          "AAD Client ID for Azure account authentication - used to authenticate to Azure for ACI creation",
+		"AZURE_CLIENT_SECRET":      "AAD Client Secret for Azure account authentication - used to authenticate to Azure for ACI creation",
+		"AZURE_TENANT_ID":          "Azure AAD Tenant Id Azure account authentication - used to authenticate to Azure for ACI creation",
+		"AZURE_SUBSCRIPTION_ID":    "Azure Subscription Id - this is the subscription to be used for ACI creation, if not specified the default subscription is used",
+		"AZURE_APP_ID":             "Azure Applications Id - this is the application to be used to authenticate to Azure",
+		"ACI_RESOURCE_GROUP":       "The name of the existing Resource Group to create the ACI instance in, if not specified a Resource Group will be created",
+		"ACI_LOCATION":             "The location to create the ACI Instance in",
+		"ACI_NAME":                 "The name of the ACI instance to create - if not specified a name will be generated",
+		"ACI_DO_NOT_DELETE":        "Do not delete RG and ACI instance created - useful for debugging - only deletes RG if it was created by the driver",
+		"ACI_MSI_TYPE":             "If this is set to user or system the created ACI Container Group will be launched with MSI",
+		"ACI_SYSTEM_MSI_ROLE":      "The role to be asssigned to System MSI User - used if ACI_MSI_TYPE == system, if this is null or empty then the role defaults to contributor",
+		"ACI_SYSTEM_MSI_SCOPE":     "The scope to apply the role to System MSI User - will attempt to set scope to the  Resource Group that the ACI Instance is being created in if not set",
+		"ACI_USER_MSI_RESOURCE_ID": "The resource Id of the MSI User - required if ACI_MSI_TYPE == User",
+	}
+}
+
+// SetConfig sets ACI driver configuration
+func (d *ACIDriver) SetConfig(settings map[string]string) {
+	d.config = settings
+}
+
+// Run executes the ACI driver
+func (d *ACIDriver) Run(op *Operation) error {
+	return d.exec(op)
+}
+
+// Handles indicates that the ACI driver supports "docker" and "oci"
+func (d *ACIDriver) Handles(dt string) bool {
+	return dt == ImageTypeDocker || dt == ImageTypeOCI
+}
+
+func (d *ACIDriver) exec(op *Operation) (reterr error) {
+	d.out = op.Out
+	d.deleteACIResources = true
+	if len(d.config["ACI_DO_NOT_DELETE"]) > 0 && strings.ToLower(d.config["ACI_DO_NOT_DELETE"]) == "true" {
+		d.deleteACIResources = false
+	}
+
+	d.log("Delete Resources:", d.deleteACIResources)
+	d.inCloudShell = len(os.Getenv("ACC_CLOUD")) > 0
+	d.log("In Cloud Shell:", d.inCloudShell)
+	if d.Simulate {
+		return nil
+	}
+
+	err := d.loginToAzure()
+	if err != nil {
+		return fmt.Errorf("cannot Login To Azure: %v", err)
```

Comment:
nit: Why don't you use `%s` here?

Reply:
There is a trend towards using `%v` for errors in case the error has some rich content that isn't expressed well in the string form of the error. I think we should probably agree a consistent approach for duffle and then stick to it...